### PR TITLE
Add support for proxying in RStudio Workbench

### PR DESCRIPTION
**Pull request by nstrayer**
This PR adds the ability for the ui editor to be proxied properly through RStudio Workbench.

### CODE REVIEW

**Comment by nstrayer** [comment]
This will fix https://github.com/rstudio/shinyuieditor/issues/69.

File: inst/ui-editor-react/src/websocket_hooks/useConnectToWebsocket.tsx [hunk 1]
```diff
@@ -54,8 +69,9 @@ export function useConnectToWebsocket() {
 
       // If we're using the dev proxy we should just go straight to websocket.
       // Otherwise use the same location as the main app
```
Comment (jcheng5) [inline inst/ui-editor-react/src/websocket_hooks/useConnectToWebsocket.tsx:71]: This comment lives in buildWebsocketPath() now
```suggestion
```

File: inst/ui-editor-react/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   },
   "scripts": {
     "start": "vite",
-    "build": "vite build",
+    "build": "vite build --base=./",
```
Comment (jcheng5) [inline inst/ui-editor-react/package.json:31]: Would it make sense to put `base: "./"` in a vite.config.js file?

**Review by jcheng5 — APPROVED** [review]
LGTM other than these nits, assuming rstudioapi is always installed on RStudio Server and Workbench.

File: R/AppPreviewClass.R [hunk 3]
```diff
@@ -163,7 +167,42 @@ get_app_url <- function(host, port) {
...
+  has_rstudioapi <- rlang::is_installed("rstudioapi")
+  running_on_hosted_service <- identical(Sys.getenv("RSTUDIO_PROGRAM_MODE"), "server")
+
+  if (!has_rstudioapi && running_on_hosted_service) {
+    # If we're on a server and can't translate the URL to a proxied one then we
+    # need to let the user know so they can install the rstudioapi package
+    stop(
+      "In order to use shinyuieditor on a hosted version of RStudio you will ",
+      "need to install the rstudioapi package.\n",
+      "Run install.packages(\"rstudioapi\") to install it."
```
Comment (jcheng5) [inline R/AppPreviewClass.R:195]: This all looks great. Just want to verify that you tried running in this configuration, and verified that 1) this error does blow all the way to the top of the stack, and 2) no resources that were already acquired by the time you got to here were left in a dangling state?

My point being, if either of those are problems or may become problems in the future, you could instead do this `!has_rstudioapi && running_on_hosted_service` proactively, way early in the call to `launch_editor`.

**Pull request merged**